src/librustc_resolve/lib.rs

```diff
@@ -117,6 +119,27 @@ enum SuggestionType {
     NotFound,
 }
 
+/// Which candidates are shown: only the accessible ones,
+/// or also the not accessible ones. By default, we only
+/// show accessible candidates (`Accessible*` cases). If no
+/// accessible candidates exist, we show not accessible ones,
+/// `NotAccessible`. If, in the worst case, no candidates
+/// are found, we mark it as `NoCandidatesAtAll`
+pub enum CandidateAccessibility {
```

Comment:
If I understand correctly, we don't need this enum anymore. Our new strategy is to collect all items in the current crate and all accessible items in external crates, so there's no need for two classes (accessible and inaccessible) of suggestion.